### PR TITLE
Remaining D2k changes to restore original balancing/gameplay

### PR DESCRIPTION
**Pull request by reaperrr**
#9458 + this should wrap up the implementation of @OmegaBolt's PR.

Don't let the number of commits intimidate you, I split everything I could into individual commits for easier reviewing.

### CODE REVIEW

File: mods/d2k/rules/defaults.yaml [hunk 1]
```diff
@@ -267,15 +273,16 @@
...
 	Guardable:
 		Range: 3c0
 	WithCrumbleOverlay:
 	Demolishable:
 	DamagedWithoutFoundation:
 	ThrowsShrapnel:
-		Weapons: Debris, Debris2, Debris3, Debris4
-		Pieces: 3, 7
-		Range: 2c0, 5c0
+		Weapons: debris, debris2, debris3, debris4
```
Comment (GraionDilach) [inline mods/d2k/rules/defaults.yaml]: Is this intended?

Comment (reaperrr) [inline mods/d2k/rules/defaults.yaml]: No, will fix it.

**Comment by reaperrr** [comment]
Updated.

- left veterancy enabled, but toned down stat gains particularly at higher levels
- removed commit that disabled AllowMovement on infantry AutoTarget
- removed commit that disabled some crate actions
- fixed debris typo

**Comment by OmegaBolt** [comment]
Why would you keep veterancy and allow movement?

As was discussed in some other commit, veterancy could be purely visual, which I think is the best idea personally so it doesn't interfere with balance and yet gives people that personal attachment to long lasting units.

Allowing movement from autotargeting is one of the absolute most annoying and stupid ideas to put in a game about unit control. The original games never did it, except in TS and RA2 where your anti-aircraft units would chase aircraft and that's commonly considered by the community an annoying bug. I do not want my infantry to ever chase down anything unless I've set it to guard mode, and even then it shouldn't hunt a unit across the whole map.

**Comment by GraionDilach** [comment]
Regarding the latter, #9611 would make it mostly redundant.

**Comment by penev92** [comment]
I'm for keeping veterancy. D2k is a mod where your units are dead meat from the moment they exit the factory anyway. Unit preservation is almost non-existent (and that's fine). So I don't think veterancy will have a big (if any) effect on the gameplay (as long as we keep the bonuses small).

File: mods/d2k/rules/defaults.yaml [hunk 2]
```diff
@@ -267,6 +272,7 @@
 	ShakeOnDeath:
 	Sellable:
 		SellSounds: BUILD1.WAV
+	AutoTargetIgnore:
```
Comment (pchote) [inline mods/d2k/rules/defaults.yaml]: Isn't this going to break the AI?

File: mods/d2k/rules/arrakis.yaml [hunk 3]
```diff
@@ -47,6 +47,7 @@ sandworm:
 		Weapon: WormJaw
 	Sandworm:
 		WanderMoveRadius: 5
+		ChanceToDisappear: 100
```
Comment (pchote) [inline mods/d2k/rules/arrakis.yaml]: It would be better to change this in the trait instead, IMO.

File: mods/d2k/rules/defaults.yaml [hunk 4]
```diff
@@ -124,13 +124,16 @@
 	Targetable:
 		TargetTypes: Ground, Vehicle
 		RequiresForceFire: yes
-	Capturable:
-		Type: husk
-		AllowAllies: yes
-		CaptureThreshold: 1.0
-	TransformOnCapture:
-		ForceHealthPercentage: 25
+#	Capturable:
```
Comment (pchote) [inline mods/d2k/rules/defaults.yaml]: Please don't leave commented out dead code in the rules.

File: mods/d2k/rules/world.yaml [hunk 5]
```diff
@@ -35,6 +35,10 @@
 		Name: Corrino
 		InternalName: corrino
 		Selectable: false
+	Faction@Smuggler:
```
Comment (pchote) [inline mods/d2k/rules/world.yaml]: This doesn't appear to be used yet, but if you want to keep it will need entries set up in the building `FactionImages` dictionaries.

Comment (pchote) [inline mods/d2k/rules/world.yaml]: IIRC smugglers should use ordos(?) artwork for everything except their custom starport, which we currently have set up as corrino.

File: mods/d2k/rules/world.yaml [hunk 6]
```diff
@@ -59,6 +63,8 @@ World:
 	ProductionQueueFromSelection:
 		ProductionPaletteWidget: PRODUCTION_PALETTE
 	WormManager:
+		Minimum: 0
+		SpawnInterval: 6000
```
Comment (pchote) [inline mods/d2k/rules/world.yaml]: The defaults could be changed instead.

File: mods/d2k/rules/structures.yaml [hunk 7]
```diff
@@ -59,8 +60,9 @@ construction_yard:
 	RevealsShroud:
 		Range: 10c0
 	Production:
-		Produces: Building, Upgrade
+		Produces: Building
 	Exit:
```
Comment (pchote) [inline mods/d2k/rules/structures.yaml]: I don't think we need this anymore, so this trait can be removed (it was only used for upgrades, which have now moved to the player).

File: mods/d2k/rules/world.yaml (hunk 5 above)
Comment (penev92) [inline mods/d2k/rules/world.yaml]: That is correct.

File: mods/d2k/rules/defaults.yaml [hunk 8]
```diff
@@ -222,7 +225,7 @@
 	Inherits@2: ^SpriteActor
 	AppearsOnRadar:
 		UseLocation: yes
-	HiddenUnderFog:
+	HiddenUnderShroud:
```
Comment (pchote) [inline mods/d2k/rules/defaults.yaml]: This will make aircraft visible to *everyone* under the fog.

File: mods/d2k/rules/world.yaml (hunk 5 above)
Comment (reaperrr) [inline mods/d2k/rules/world.yaml]: Are smugglers the same as mercenaries? IIRC the AI-controlled faction allied to Ordos in the campaign had a different artwork for their heavy factory, too.

Comment (pchote) [inline mods/d2k/rules/world.yaml]: Nope, mercenaries were yet another faction that copied one of the base factions for everything except their custom heavy factory.

Comment (pchote) [inline mods/d2k/rules/world.yaml]: IMO it would make sense to have a dedicated PR for setting up the smuggers/mercenaries(/fremen?) factions properly.

Comment (penev92) [inline mods/d2k/rules/world.yaml]: Smugglers used Ordos units and structures and had their own Starport.
Mercenaries used Ordos units and structures and had their own Heavy Factory.
Fremen used Atreides units and structures.
House Corrino (the Imperial family) used Harkonnen units and structures and had their own Palace.

Might as well leave those for a separate PR.

File: mods/d2k/rules/defaults.yaml (hunk 2 above)
Comment (reaperrr) [inline mods/d2k/rules/defaults.yaml]: The AI still attacked my buildings normally in a test game I had.

File: mods/d2k/rules/defaults.yaml (hunk 8 above)
Comment (reaperrr) [inline mods/d2k/rules/defaults.yaml]: IIRC the main reason @OmegaBolt did this was that Carryalls and Ornithopters would otherwise be much harder to shoot down than in the original.

**Comment by reaperrr** [comment]
Updated.

Not sure about the aircraft HiddenUnderFog, should I leave that in or remove it?

File: mods/d2k/rules/defaults.yaml (hunk 8 above)
Comment (pchote) [inline mods/d2k/rules/defaults.yaml]: I assumed that it was done to make them visible to their owner under the fog.  That should be done by removing the `AlwaysVisibleStances` line below instead.

**Comment by reaperrr** [comment]
Removed the aircraft visibility change. I guess we should let people play some games, if lots of them complain we can still change it.

File: mods/d2k/rules/defaults.yaml (hunk 2 above)
Comment (pchote) [inline mods/d2k/rules/defaults.yaml]: Strange.  AFAIK we haven't changed anything relevant to AutoTarget or the AI since the failure from #3337 / #3467.

File: mods/d2k/rules/world.yaml (hunk 5 above)
Comment (MustaphaTR) [inline mods/d2k/rules/world.yaml]: According to Tibed in orginal game Corrino has own CYard and War Factory. But they are same as Harkonnen ones as graphics and code. Possibly used in missions to separate them from Harkonnen ones.

**Comment by penev92** [comment]
- [ ] Please include mission `atreides 02b` to the first commit as well.
 - [ ] I am now able to build any structure apart from concrete slabs without a ConYard.

File: mods/d2k/maps/atreides-01a/map.yaml [hunk 9]
```diff
@@ -145,8 +147,8 @@ Rules:
 			Minimum: 1
 			Maximum: 1
 	construction_yard:
-		Production:
-			Produces: Building
+		Buildable:
+			Prerequisites: ~disabled
```
Comment (abcdefg30) [inline mods/d2k/maps/atreides-01a/map.yaml]: Shouldn't `construction_yard` be unabuildable by default?
If so, please remove this (also from the other missions).

**Comment by reaperrr** [comment]
@penev92 @abcdefg30 fixed prerequisites and missions.

**Comment by pchote** [comment]
Changes look ok, and didn't notice anything that looked too bogus while spectating a bot match with these changes, so :+1:.

File: mods/d2k/rules/defaults.yaml (hunk 2 above)
Comment (penev92) [inline mods/d2k/rules/defaults.yaml]: But this *does* break scripted reinforcements in the missions, which can sit idly by my ConYard, chilling.
I think it would be simplest to just not change that here. Can be done in a later PR if you want.

**Comment by penev92** [comment]
Adding "Changes requested" for building `AutoTargetIgnore` breaking scripted reinforcements.

**Comment by reaperrr** [comment]
Updated.

- removed AutoTargetIngore commit
- removed warhead falloff changes from debris commit, this is properly taken care of in my warhead PR now

File: mods/d2k/maps/atreides-01a/map.yaml [hunk 10]
```diff
@@ -134,6 +134,8 @@ Rules:
 		-ConquestVictoryConditions:
 		MissionObjectives:
 			EarlyGameOver: true
+		Production:
+			Produces: Building
```
Comment (penev92) [inline mods/d2k/maps/atreides-01a/map.yaml]: Is this still necessary? Selling my ConYard has an odd side-effect.
While that's not an issue, you need to just exclude Upgrades here, not move Building to the player, right?

File: mods/d2k/rules/arrakis.yaml [hunk 11]
```diff
@@ -32,11 +32,11 @@ sandworm:
 		Description: Attracted by vibrations in the sand.\nWill eat units whole and has a large appetite.
 	Health:
 		HP: 1200
```
Comment (penev92) [inline mods/d2k/rules/arrakis.yaml]: Partially related: it takes 2 trikes to kill a worm before it can eat one of them.
TibEd says Sandworm has 9990 HP, so this definitely needs a buff.

**Comment by reaperrr** [comment]
Updated.

- Fixed Production issue
- Increased worm HP, but increased target rescan delay from 32 to 125 ticks and reduced search radius from 27 to 20 cells

File: mods/d2k/rules/structures.yaml [hunk 12]
```diff
@@ -895,9 +895,6 @@ conyard.harkonnen:
 conyard.ordos:
 	Inherits: construction_yard
 	Buildable:
-		Queue: Building
-		BuildPaletteOrder: 1000
-		Prerequisites: ~disabled
```
Comment (Mailaender) [inline mods/d2k/rules/structures.yaml]: I guess this makes the construction yard appear twice with cheats on:

image

**Comment by reaperrr** [comment]
Updated, fixed the cyard double appearance.

**Comment by penev92** [comment]
Adding "Changes requested" for the first commit as discussed on IRC.

**Comment by reaperrr** [comment]
Updated.

**Comment by obrakmann** [comment]
I'll relieve this one of its misery as well. :+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/9107ff42356ced738edf4a126be541fc8d9b2eaa)
